src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -647,10 +709,69 @@ class Diagram extends Component<Props> {
     return target && target.model instanceof RouterNodeModel
   }
 
+  updateNodeContent(data) {
+    const {
+      node: { contents },
+      index
+    } = this.state.editingNodeContent
+    const newContents = [...contents]
+
+    if (
+      newContents[index][this.state.currentLang]?.contentType &&
+      newContents[index][this.state.currentLang].contentType !== data.contentType
+    ) {
+      newContents[index] = { [this.state.currentLang]: data }
+    } else {
+      newContents[index][this.state.currentLang] = data
+    }
```

Comment:
Can you refactor this part ? Those lines are kind of long, also the first verification for `contentType` is not necessary

Reply:
Will refactor, but it is necessary otherwise it compares null to a content type from the data and it will reset the other langs everytime you add a new one